### ISSUE TITLE
After 95e16f7 commit I cannot anymore generate FLUX 16fp + one LoRA

### ISSUE THREAD

**Issue by DavideAlidosi** [lllyasviel/stable-diffusion-webui-forge#1607]
After updating to commit 95e16f7 or later I am no longer able to generate images using the FP16 version of FLUX Dev with one or more loaded LoRAs. This happens due to higher VRAM consumption at the same settings.
Revert back to commint c70fb38 everything goes back to working as previously.
Thanks for support.

**Comment by DavideAlidosi** [comment]
> do you have full log from beginning
Sure!
commit 95e16f7:
```
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Version: f2.0.1v1.10.1-previous-471-g1e0e861b
Commit hash: 1e0e861b0de4b19326acfdbcb11a656adca8d57d
Launching Web UI with arguments: --cuda-malloc --disable-gpu-warning
Using cudaMallocAsync backend.
Total VRAM 24575 MB, total RAM 65467 MB
pytorch version: 2.3.1+cu121
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 3090 : cudaMallocAsync
VAE dtype preferences: [torch.bfloat16, torch.float32] -> torch.bfloat16
CUDA Using Stream: False
L:\indrema-forge_FLUX-alpha.0.2\system\python\lib\site-packages\transformers\utils\hub.py:127: FutureWarning: Using `TRANSFORMERS_CACHE` is deprecated and will be removed in v5 of Transformers. Use `HF_HOME` instead.
  warnings.warn(
Using pytorch cross attention
Using pytorch attention for VAE
ControlNet preprocessor location: L:\indrema-forge_FLUX-alpha.0.2\webui\models\ControlNetPreprocessor
2024-08-30 16:33:10,550 - ControlNet - INFO - ControlNet UI callback registered.
Model selected: {'checkpoint_info': {'filename': 'L:\\indrema-forge_FLUX-alpha.0.2\\webui\\models\\Stable-diffusion\\flux1-dev.safetensors', 'hash': 'b04b3ba1'}, 'additional_modules': ['L:\\indrema-forge_FLUX-alpha.0.2\\webui\\models\\VAE\\ae.safetensors', 'L:\\indrema-forge_FLUX-alpha.0.2\\webui\\models\\text_encoder\\clip_l.safetensors', 'L:\\indrema-forge_FLUX-alpha.0.2\\webui\\models\\text_encoder\\t5xxl_fp16.safetensors'], 'unet_storage_dtype': None}
Using online LoRAs in FP16: False
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
Startup time: 23.3s (prepare environment: 3.5s, import torch: 9.0s, initialize shared: 0.4s, other imports: 1.2s, load scripts: 4.0s, create ui: 3.3s, gradio launch: 1.6s).
Environment vars changed: {'stream': False, 'inference_memory': 1024.0, 'pin_shared_memory': False}
------------------
[Low VRAM Warning] You just set Forge to use 100% GPU memory (23551.00 MB) to load model weights.
[Low VRAM Warning] This means you will have 0% GPU memory (0.00 MB) to do matrix computation. Computations may fallback to CPU or go Out of Memory.
[Low VRAM Warning] In many cases, image generation will be 10x slower.
[Low VRAM Warning] Make sure that you know what you are testing.
------------------
Error parsing "sv_negative: "
Loading Model: {'checkpoint_info': {'filename': 'L:\\indrema-forge_FLUX-alpha.0.2\\webui\\models\\Stable-diffusion\\flux1-dev.safetensors', 'hash': 'b04b3ba1'}, 'additional_modules': ['L:\\indrema-forge_FLUX-alpha.0.2\\webui\\models\\VAE\\ae.safetensors', 'L:\\indrema-forge_FLUX-alpha.0.2\\webui\\models\\text_encoder\\clip_l.safetensors', 'L:\\indrema-forge_FLUX-alpha.0.2\\webui\\models\\text_encoder\\t5xxl_fp16.safetensors'], 'unet_storage_dtype': None}
[Unload] Trying to free all memory for cuda:0 with 0 models keep loaded ...
StateDict Keys: {'transformer': 780, 'vae': 244, 'text_encoder': 196, 'text_encoder_2': 220, 'ignore': 0}
Using Default T5 Data Type: torch.float16
Working with z of shape (1, 16, 32, 32) = 16384 dimensions.
K-Model Created: {'storage_dtype': torch.bfloat16, 'computation_dtype': torch.bfloat16}
Model loaded in 1.0s (unload existing model: 0.2s, forge model load: 0.8s).
[LORA] Loaded L:\indrema-forge_FLUX-alpha.0.2\webui\models\Lora\persone\Giulia_Mancini.safetensors for KModel-UNet with 304 keys at weight 1.0 (skipped 0 keys)
Skipping unconditional conditioning when CFG = 1. Negative Prompts are ignored.
To load target model JointTextEncoder
Begin to load 1 model
[Unload] Trying to free 13464.34 MB for cuda:0 with 0 models keep loaded ...
[Memory Management] Current Free GPU Memory: 23293.00 MB
[Memory Management] Required Model Memory: 9569.49 MB
[Memory Management] Required Inference Memory: 1024.00 MB
[Memory Management] Estimated Remaining GPU Memory: 12699.51 MB
Moving model(s) has taken 6.13 seconds
Distilled CFG Scale: 2.5
To load target model KModel
Begin to load 1 model
[Unload] Trying to free 30800.42 MB for cuda:0 with 0 models keep loaded ...
[Unload] Current free memory is 13581.68 MB ...
[Unload] Unload model JointTextEncoder
[Memory Management] Current Free GPU Memory: 23223.66 MB
[Memory Management] Required Model Memory: 22700.13 MB
[Memory Management] Required Inference Memory: 1024.00 MB
[Memory Management] Estimated Remaining GPU Memory: -500.47 MB
Patching LoRA by precomputing model weights.
Patching LoRAs for KModel: 100%|█████████████████████████████████████████████████████| 304/304 [00:52<00:00,  5.79it/s]
LoRA patching has taken 52.50 seconds
[Memory Management] Loaded to CPU Swap: 1890.00 MB (blocked method)
[Memory Management] Loaded to GPU: 20810.13 MB
Moving model(s) has taken 62.87 seconds
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:32<00:00,  1.65s/it]
To load target model IntegratedAutoencoderKL███████████████████████████████████████████| 20/20 [00:31<00:00,  1.66s/it]
Begin to load 1 model
[Unload] Trying to free 4495.77 MB for cuda:0 with 0 models keep loaded ...
[Unload] Current free memory is 2389.43 MB ...
[Unload] Unload model KModel

```
Commint 95e16f7:
```
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Version: f2.0.1v1.10.1-previous-471-g1e0e861b
Commit hash: 1e0e861b0de4b19326acfdbcb11a656adca8d57d
Launching Web UI with arguments: --cuda-malloc --disable-gpu-warning
Using cudaMallocAsync backend.
Total VRAM 24575 MB, total RAM 65467 MB
pytorch version: 2.3.1+cu121
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 3090 : cudaMallocAsync
VAE dtype preferences: [torch.bfloat16, torch.float32] -> torch.bfloat16
CUDA Using Stream: False
L:\indrema-forge_FLUX-alpha.0.2\system\python\lib\site-packages\transformers\utils\hub.py:127: FutureWarning: Using `TRANSFORMERS_CACHE` is deprecated and will be removed in v5 of Transformers. Use `HF_HOME` instead.
  warnings.warn(
Using pytorch cross attention
Using pytorch attention for VAE
ControlNet preprocessor location: L:\indrema-forge_FLUX-alpha.0.2\webui\models\ControlNetPreprocessor
2024-08-30 18:28:58,616 - ControlNet - INFO - ControlNet UI callback registered.
Model selected: {'checkpoint_info': {'filename': 'L:\\indrema-forge_FLUX-alpha.0.2\\webui\\models\\Stable-diffusion\\flux1-dev.safetensors', 'hash': 'b04b3ba1'}, 'additional_modules': ['L:\\indrema-forge_FLUX-alpha.0.2\\webui\\models\\VAE\\ae.safetensors', 'L:\\indrema-forge_FLUX-alpha.0.2\\webui\\models\\text_encoder\\clip_l.safetensors', 'L:\\indrema-forge_FLUX-alpha.0.2\\webui\\models\\text_encoder\\t5xxl_fp16.safetensors'], 'unet_storage_dtype': None}
Using online LoRAs in FP16: False
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
Startup time: 28.0s (prepare environment: 5.0s, import torch: 11.1s, initialize shared: 0.5s, other imports: 1.6s, load scripts: 4.5s, create ui: 3.3s, gradio launch: 1.6s).
Environment vars changed: {'stream': False, 'inference_memory': 1024.0, 'pin_shared_memory': False}
------------------
[Low VRAM Warning] You just set Forge to use 100% GPU memory (23551.00 MB) to load model weights.
[Low VRAM Warning] This means you will have 0% GPU memory (0.00 MB) to do matrix computation. Computations may fallback to CPU or go Out of Memory.
[Low VRAM Warning] In many cases, image generation will be 10x slower.
[Low VRAM Warning] Make sure that you know what you are testing.
------------------
Error parsing "sv_negative: "
Loading Model: {'checkpoint_info': {'filename': 'L:\\indrema-forge_FLUX-alpha.0.2\\webui\\models\\Stable-diffusion\\flux1-dev.safetensors', 'hash': 'b04b3ba1'}, 'additional_modules': ['L:\\indrema-forge_FLUX-alpha.0.2\\webui\\models\\VAE\\ae.safetensors', 'L:\\indrema-forge_FLUX-alpha.0.2\\webui\\models\\text_encoder\\clip_l.safetensors', 'L:\\indrema-forge_FLUX-alpha.0.2\\webui\\models\\text_encoder\\t5xxl_fp16.safetensors'], 'unet_storage_dtype': None}
[Unload] Trying to free all memory for cuda:0 with 0 models keep loaded ... Done.
StateDict Keys: {'transformer': 780, 'vae': 244, 'text_encoder': 196, 'text_encoder_2': 220, 'ignore': 0}
Using Default T5 Data Type: torch.float16
Working with z of shape (1, 16, 32, 32) = 16384 dimensions.
K-Model Created: {'storage_dtype': torch.bfloat16, 'computation_dtype': torch.bfloat16}
Model loaded in 1.4s (unload existing model: 0.2s, forge model load: 1.2s).
[LORA] Loaded L:\indrema-forge_FLUX-alpha.0.2\webui\models\Lora\persone\Giulia_Mancini.safetensors for KModel-UNet with 304 keys at weight 1.0 (skipped 0 keys)
Skipping unconditional conditioning when CFG = 1. Negative Prompts are ignored.
[Unload] Trying to free 13464.34 MB for cuda:0 with 0 models keep loaded ... Done.
[Memory Management] Target: JointTextEncoder, Free GPU: 23293.00 MB, Model Require: 9569.49 MB, Inference Require: 1024.00 MB, Remaining: 12699.51 MB, All loaded to GPU.
Moving model(s) has taken 9.86 seconds
Distilled CFG Scale: 2.5
[Unload] Trying to free 30800.42 MB for cuda:0 with 0 models keep loaded ... Current free memory is 13581.68 MB ... Unload model JointTextEncoder Done.
[Memory Management] Target: KModel, Free GPU: 23223.66 MB, Model Require: 22700.13 MB, Inference Require: 1024.00 MB, Remaining: -500.47 MB, CPU Swap Loaded (blocked method): 1890.00 MB, GPU Loaded: 20810.13 MB
Patched LoRAs by precomputing model weights; Moving model(s) has taken 53.40 seconds
 15%|████████████████████████████████████▊                                                                                                                                                                                                                | 3/20 [01:09<06:29, 22.89s/it]
Total progress:  15%|██████████████████████████████████▎                                                                                                                                                                                                  | 3/20 [00:45<04:31, 15.95s/it]
```
Also at the end of computation the last commit give me a crash.

**Comment by DavideAlidosi** [comment]
In the terminal the commit number is the same because I downgraded ForgeUI by downloading the package of the old version from the site as a .zip

**Comment by lllyasviel** [comment]
Update and follow the instruction in your console.

**Comment by DavideAlidosi** [comment]
Hi lllyasviel, thank you for responding. I haven't updated yet, but seen the code you added in one of the last commits related to warnings.
The probblema that I reported from my point of view cannot be blamed on a misunderstanding of mine, I understand by myself that I have always been at the VRAM limit, but the fact remains that with an old commit the generation works and has worked for days, allowing me to make countless images.
That you then decided to make an update that changed, I assume with the best of intentions, the handling of VRAM I can accept, but otherwise in my opinion it is a regression. Also because if not, it doesn't explain why from one moment to the next the system is asked for about 4GB more VRAM.

**Comment by lllyasviel** [comment]
Because the previous version is same as using GPU Weights as 0 when patching loras and you always waste entire 24GB vram.

New version allows you to really use GPU with the value on UI when patching loras

**Comment by DavideAlidosi** [comment]
I upgraded to the latest commit, lowered the GPU Weight to 22000, compared to the old commit the generation is 10% slower, but just before saving the image I restarted explorer and crashed CMD.
I maintain that something is wrong with the latest versions, in any case I will do yet another downgrade.... At least I'm learning how to use Git better.

**Comment by lllyasviel** [comment]
if it crashes then it has nothing to do with updates. 

If it crashes at one version, then no matter what version you use, it will always crash at more or less loras at all versions. Maybe one version will crash at 2 lora while another version will crash at 3 lora or cn or so

The correct way to fix crash is to really read document and really follow document rather than being so confident to ignore it:

https://github.com/lllyasviel/stable-diffusion-webui-forge/discussions/1474

**Comment by DavideAlidosi** [comment]
I understand your point of view, by lowering the value of GPU Weights (MB) I can finish the generations although with a noticeable loss of performance. Probably this update was necessary and maybe they are on a small segment of users can bring negative effects, and I understand that.
Thanks anyway for the support.
